app/helpers/duplicate_intake_guard.rb

```diff
@@ -1,17 +1,9 @@
 class DuplicateIntakeGuard < SimpleDelegator
   def has_duplicate?
-    ClientLoginService.new(:gyr).accessible_intakes
-      .where(
-         (arel_table[:email_address].eq(email_address)
-           .and(arel_table[:email_address].not_eq(nil)))
-         .or(arel_table[:phone_number].eq(phone_number)
-           .and(arel_table[:phone_number].not_eq(nil)))
-      ).exists?
-  end
-
-  private
+    ClientLoginService.new(:gyr).accessible_intakes.each do |intake|
+      return true if DeduplificationService.duplicates(intake, :hashed_primary_ssn).exists?
```

Comment:
Yay! I love that the deduplification sevice could be useful here.. However, this code would loop through every accessible intake in the application, and there will eventually potentially be millions! I think we need to refactor the Deduplication service query so that we can chain it onto accessible intakes and make these guys compatible. We can pair on that if you'd like?

Reply:
@bytheway875 Oh yeah just saw this comment, thank you! But yes I caught this and have been trying to resolve it by doing this 

```
dupes = DeduplificationService.duplicates(intake, :hashed_primary_ssn)

if intake.type == "Intake::GyrIntake" && intake.drop_off?
      dupes.exist?
    elsif intake.type == "Intake::GyrIntake" && !intake.drop_off?
      dupes.exists?(primary_consented_to_service: "yes")
    else
      dupes.exists?('sms_phone_number_verified_at != nil OR email_address_verified_at != nil OR navigator_has_verified_client_identity != nil')
    end
```

Reply:
ugh but have a failing test because its not creating tax returns on my intake for some reason 🙄😭

Reply:
@em i think we can make something work that is a little more straightforward.

Reply:
Ooo yeah i updated it

Reply:
maybe this could be more straight forward too
```
def has_duplicate?
    intake = Intake.find(id)
    return if intake.primary_ssn == nil

    dupes = DeduplificationService.duplicates(intake, :hashed_primary_ssn)

    if intake.type == "Intake::GyrIntake"
      dupes.any?(&:drop_off?) ? dupes.exists? : dupes.exists?(primary_consented_to_service: "yes")
    else
      dupes.exists?('sms_phone_number_verified_at != nil OR email_address_verified_at != nil OR navigator_has_verified_client_identity != nil')
    end
  end
```